contracts/test/OracleMainnet.t.sol

```diff
@@ -2042,58 +2074,76 @@ contract OraclesMainnet is TestAccounts {
 
     // --- Low gas market oracle reverts ---
 
-    // --- Call these functions with 10k gas - i.e. enough to run out of gas in the Chainlink calls ---
     function testRevertLowGasSTETHOracle() public {
+        // Confirm call to the real external contracts succeeds with sufficient gas i.e. 500k
+        (bool success,) = address(wstethPriceFeed).call{gas: 500000}(abi.encodeWithSignature("fetchPrice()"));
+        assertTrue(success);
+
+        // Etch gas guzzler to the oracle
+        etchGasGuzzlerToStethOracle(address(gasGuzzlerOracle).code);
+
+        // After etching the gas guzzler to the oracle, confirm the same call with 500k gas now reverts due to OOG
         vm.expectRevert(MainnetPriceFeedBase.InsufficientGasForExternalCall.selector);
-        // just catch return val to suppress warning
-        (bool success,) = address(wstethPriceFeed).call{gas: 10000}(abi.encodeWithSignature("fetchPrice()"));
-        assertFalse(success);
+        (bool revertAsExpected,) = address(wstethPriceFeed).call{gas: 500000}(abi.encodeWithSignature("fetchPrice()"));
+        assertTrue(revertAsExpected);
```

Comment:
Interesting, so when the failure is catched by `vm.expecRevert` then `bool` return value is true?

Reply:
Yes, strange design choice in Foundry IMO. See the "Gotcha: low level calls" section here:
https://book.getfoundry.sh/cheatcodes/expect-revert